app/channel.go

```diff
@@ -71,7 +71,8 @@ func (a *App) JoinDefaultChannels(teamId string, user *model.User, shouldBeAdmin
 			cm := &model.ChannelMember{
 				ChannelId:   channel.Id,
 				UserId:      user.Id,
-				SchemeUser:  true,
+				SchemeGuest: user.IsGuest(),
```

Comment:
When I'm a guest and I join to a channel (this is a bad example because in theory I don't get joined to the default channels) I get guest membership not user membership

Reply:
Makes sense.